File: test-bridge-validate.js
const fs = require("fs");
const crypto = require("crypto");
const https = require("https");

// 读取 secret
const envContent = fs.readFileSync(".env.local", "utf8");
const lines = envContent.split("\n");
let secret = null;

for (const line of lines) {
  const trimmed = line.trim();
  if (trimmed.startsWith("CONTENTOPS_BRIDGE_SECRET=*** {
    const parts = trimmed.split("=");
    if (parts.length === 2) {
      secret = parts[1];
      break;
    }
  }
}

if (!secret) {
  console.log("bridgeValidateOnlyPass=false");
  console.log("bridgeValidateOnlyErrorType=missing_secret");
  process.exit(1);
}

// 构造请求体
const payload = {
  contentType: "topic",
  title: "Test Topic",
  gatewayLocation: "local_mac",
  localHermesRunId: "test-run-" + Date.now(),
  planningUsed: true,
  fallbackUsed: false,
  validateOnly: true
};

const body = JSON.stringify(payload);

// 生成 HMAC 签名
const signature = crypto.createHmac("sha256", secret).update(body).digest("hex");

// 发送请求
const options = {
  hostname: "jueshi.net",
  port: 443,
  path: "/api/internal/contentops/drafts",
  method: "POST",
  headers: {
    "Content-Type": "application/json",
    "x-contentops-signature": signature,
    "Content-Length": Buffer.byteLength(body)
  }
};

const req = https.request(options, (res) => {
  let data = "";
  res.on("data", (chunk) => { data += chunk; });
  res.on("end", () => {
    if (res.statusCode === 200) {
      const response = JSON.parse(data);
      if (response.success) {
        console.log("bridgeValidateOnlyPass=true");
        console.log("hmacPass=true");
        console.log("schemaPass=true");
        console.log("bridgeValidateOnlyStatus=success");
        console.log("bridgeValidateOnlyErrorType=none");
      } else {
        console.log("bridgeValidateOnlyPass=false");
        console.log("hmacPass=true");
        console.log("schemaPass=false");
        console.log("bridgeValidateOnlyStatus=schema_validation_failed");
        console.log("bridgeValidateOnlyErrorType=schema_validation_failed");
      }
    } else if (res.statusCode === 401) {
      console.log("bridgeValidateOnlyPass=false");
      console.log("hmacPass=false");
      console.log("schemaPass=false");
      console.log("bridgeValidateOnlyStatus=invalid_signature");
      console.log("bridgeValidateOnlyErrorType=invalid_signature_format");
    } else {
      console.log("bridgeValidateOnlyPass=false");
      console.log("hmacPass=unknown");
      console.log("schemaPass=unknown");
      console.log("bridgeValidateOnlyStatus=route_error");
      console.log("bridgeValidateOnlyErrorType=route_error");
      console.log("statusCode=" + res.statusCode);
    }
  });
});

req.on("error", (e) => {
  console.log("bridgeValidateOnlyPass=false");
  console.log("bridgeValidateOnlyErrorType=route_error");
  console.log("error=" + e.message);
});

req.write(body);
req.end();

File: test-bridge-validate.py
import json
import hmac
import hashlib
import requests

# 读取 secret
with open(".env.local") as f:
    for line in f:
        line = line.strip()
        if line.startswith("CONTENTOPS_BRIDGE_SECRET=*** {
            parts = line.split("=", 1)
            if len(parts) == 2:
                secret = parts[1]
                break
    else:
        print("bridgeValidateOnlyPass=false")
        print("bridgeValidateOnlyErrorType=missing_secret")
        exit(1)

# 构造请求体
payload = {
    "contentType": "topic",
    "title": "Test Topic",
    "gatewayLocation": "local_mac",
    "localHermesRunId": "test-run-001",
    "planningUsed": True,
    "fallbackUsed": False,
    "validateOnly": True
}

body = json.dumps(payload)

# 生成 HMAC 签名
signature = hmac.new(secret.encode(), body.encode(), hashlib.sha256).hexdigest()

# 发送请求
try:
    response = requests.post(
        "https://jueshi.net/api/internal/contentops/drafts",
        headers={
            "Content-Type": "application/json",
            "x-contentops-signature": signature
        },
        data=body,
        timeout=10
    )
    
    if response.status_code == 200:
        data = response.json()
        if data.get("success"):
            print("bridgeValidateOnlyPass=true")
            print("hmacPass=true")
            print("schemaPass=true")
            print("bridgeValidateOnlyStatus=success")
            print("bridgeValidateOnlyErrorType=none")
        else:
            print("bridgeValidateOnlyPass=false")
            print("hmacPass=true")
            print("schemaPass=false")
            print("bridgeValidateOnlyStatus=schema_validation_failed")
            print("bridgeValidateOnlyErrorType=schema_validation_failed")
    elif response.status_code == 401:
        print("bridgeValidateOnlyPass=false")
        print("hmacPass=false")
        print("schemaPass=false")
        print("bridgeValidateOnlyStatus=invalid_signature")
        print("bridgeValidateOnlyErrorType=invalid_signature_format")
    else:
        print("bridgeValidateOnlyPass=false")
        print("hmacPass=unknown")
        print("schemaPass=unknown")
        print("bridgeValidateOnlyStatus=route_error")
        print("bridgeValidateOnlyErrorType=route_error")
        print("statusCode=" + str(response.status_code))
except Exception as e:
    print("bridgeValidateOnlyPass=false")
    print("bridgeValidateOnlyErrorType=route_error")
    print("error=" + str(e))
